upgrade-legacy-d7.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+echo "This script will convert your Drupal 7 project to be compatible with composer, lando and Silta."
+echo "Make sure you are in the git root of your project."
+
+echo "Checking dependencies"
+git --version
+composer --version
+lando version
+json_pp -v
+
+# Use the current directory as the project name
+CURRENT_PATH=`pwd`
+PROJECT_NAME=`basename $CURRENT_PATH`
+
+# Update build.sh
+curl -s https://raw.githubusercontent.com/wunderio/WunderTools/drupal7/build.sh > build.sh
+curl -s https://raw.githubusercontent.com/wunderio/WunderTools/drupal7/drupal/build.sh > drupal/build.sh
+
+# Update site.yml
+mv drupal/conf/site.yml drupal/conf/site.yml.old
+curl -s https://raw.githubusercontent.com/wunderio/WunderTools/drupal7/drupal/conf/site.yml > drupal/conf/site.yml
+
+# Download lando configuration file, renaming the project on the way.
+curl -s https://raw.githubusercontent.com/wunderio/WunderTools/drupal7/drupal/.lando.yml | sed "s/wundertools/$PROJECT_NAME/" > drupal/.lando.yml
+
+# Download the base composer file.
+curl -s https://raw.githubusercontent.com/drupal-composer/drupal-project/7.x/composer.json > drupal/composer.json
+curl -s https://raw.githubusercontent.com/drupal-composer/drupal-project/7.x/.gitignore > drupal/.gitignore
+mkdir -p drupal/scripts/composer
+curl -s https://raw.githubusercontent.com/drupal-composer/drupal-project/7.x/scripts/composer/ScriptHandler.php > drupal/scripts/composer/ScriptHandler.php
+mkdir -p drupal/drush
+curl -s https://raw.githubusercontent.com/drupal-composer/drupal-project/7.x/drush/policy.drush.inc > drupal/drush/policy.drush.inc
+
+# Run composer install so we have the scaffolding in place.
+composer install --working-dir=drupal --no-suggest --ignore-platform-reqs
+rm drupal/composer.lock
+mv drupal/composer.json drupal/base-composer.json
+
+# Move drush aliases
+mkdir -p drupal/drush
+git mv `find drupal/conf | grep drushrc` drupal/drush/
+
+# Move custom code to the right place
+git mv drupal/code/modules/custom drupal/web/sites/all/modules/custom
+git mv drupal/code/modules/features drupal/web/sites/all/modules/features
+rmdir drupal/code/modules
+
+git mv drupal/code/themes/custom drupal/web/sites/all/themes/custom
+rmdir drupal/code/themes
+
+git mv drupal/code/profiles drupal/web/sites/all/profiles
+
+# Move patches to the right place
+if [ -d drupal/conf/patches ] ; then
+  git mv drupal/conf/patches drupal/patches
+fi
+if [ -d drupal/code/patches ] ; then
+  git mv drupal/code/patches drupal/patches
+fi
+
+if [ -d drupal/patches ] ; then
+  printf "\e[33mPlease make sure that the following patches are included in composer.json and are still relevant.\e[0m\n"
+  ls -l drupal/patches
+fi
+
+# Remove the code folder.
+# This should be empty at this point, otherwise a failure is intentional.
+rmdir drupal/code
+
```

Comment:
Should we add setting files from wundertools automatically in here? The setting files in wundertools/drupal/web/sites/defauilt/* are pretty sane and the developer can then extend them with settings.local.php or make modifications into the existing ones to make them work with the project (settings.lando.php).

We could even modify the lando setting files automatically to something and then just make a note after the script has been run to notify the developer that these need to be looked at.

Reply:
Additionally we have a script for db syncing, which is meant to be a "barebone" and somewhat working for "most" drupal7 project hosted in UC. But it can be modified by the developer of the project to be anything.

But the basic gist is, that the script has been added into lando tooling so that developers can easily call "lando syncdb" and it would automatically fetch the proper database, do anything we want for it and then import it.

?

Reply:
Weve also added an .env file https://github.com/wunderio/WunderTools/blob/drupal7/drupal/.env for environment variables, they could be even automated at somepoint, but the point is to provide a base for lando environment variables so that people can automate theyr own projects more easily with these variables. :O